### DISCUSSION TITLE
When to use ruff format versus ruff --fix

### DISCUSSION THREAD

**Discussion by paulf81** [astral-sh/ruff#9048] [Q&A] [answered]
I've been using ruff and ruff format for cleaning up python code, thank you for the excellent tools!  I had a question, I notice that running ruff format corrects some issues, while ruff --fix seems to address others.  For example, sorting the imports seems to be done by ruff --fix but not ruff format.  Is the correct usage to run both in succession?

**Comment by zanieb** [comment, marked as answer]
Hi! `ruff --fix` is an alias for `ruff check --fix` which is for the ruff _linter_. `ruff format` uses the ruff _formatter_. It may be helpful to read the linter / formatter documentation to understand the differences. For now, yeah you should run them both. In the future, we'll have a command that does both https://github.com/astral-sh/ruff/issues/8232

**Reply by paulf81** [reply]
This is very helpful, thank you!
